src/mpi/coll/src/coll_impl.c

```diff
@@ -584,60 +584,88 @@ int MPII_Coll_init(void)
         MPIR_Ineighbor_allgather_inter_algo_choice = MPIR_INEIGHBOR_ALLGATHER_INTER_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLGATHER_INTER_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_allgather_inter_algo_choice =
+            MPIR_INEIGHBOR_ALLGATHER_INTER_ALGO_GENTRAN_LINEAR;
     else
         MPIR_Ineighbor_allgather_inter_algo_choice = MPIR_INEIGHBOR_ALLGATHER_INTER_ALGO_AUTO;
 
     /* Ineighbor_allgatherv Intra */
     if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLGATHERV_INTRA_ALGORITHM, "linear"))
         MPIR_Ineighbor_allgatherv_intra_algo_choice = MPIR_INEIGHBOR_ALLGATHERV_INTRA_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLGATHERV_INTRA_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_allgatherv_intra_algo_choice =
+            MPIR_INEIGHBOR_ALLGATHERV_INTRA_ALGO_GENTRAN_LINEAR;
     else
         MPIR_Ineighbor_allgatherv_intra_algo_choice = MPIR_INEIGHBOR_ALLGATHERV_INTRA_ALGO_AUTO;
 
     /* Ineighbor_allgatherv Inter */
     if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLGATHERV_INTER_ALGORITHM, "linear"))
         MPIR_Ineighbor_allgatherv_inter_algo_choice = MPIR_INEIGHBOR_ALLGATHERV_INTER_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLGATHERV_INTER_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_allgatherv_inter_algo_choice =
+            MPIR_INEIGHBOR_ALLGATHERV_INTER_ALGO_GENTRAN_LINEAR;
     else
         MPIR_Ineighbor_allgatherv_inter_algo_choice = MPIR_INEIGHBOR_ALLGATHERV_INTER_ALGO_AUTO;
 
     /* Ineighbor_alltoall Intra */
     if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALL_INTRA_ALGORITHM, "linear"))
         MPIR_Ineighbor_alltoall_intra_algo_choice = MPIR_INEIGHBOR_ALLTOALL_INTRA_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALL_INTRA_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_alltoall_intra_algo_choice =
+            MPIR_INEIGHBOR_ALLTOALL_INTRA_ALGO_GENTRAN_LINEAR;
     else
         MPIR_Ineighbor_alltoall_intra_algo_choice = MPIR_INEIGHBOR_ALLTOALL_INTRA_ALGO_AUTO;
 
     /* Ineighbor_alltoall Inter */
     if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALL_INTER_ALGORITHM, "linear"))
         MPIR_Ineighbor_alltoall_inter_algo_choice = MPIR_INEIGHBOR_ALLTOALL_INTER_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALL_INTER_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_alltoall_inter_algo_choice =
+            MPIR_INEIGHBOR_ALLTOALL_INTER_ALGO_GENTRAN_LINEAR;
     else
         MPIR_Ineighbor_alltoall_inter_algo_choice = MPIR_INEIGHBOR_ALLTOALL_INTER_ALGO_AUTO;
 
     /* Ineighbor_alltoallv Intra */
     if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALLV_INTRA_ALGORITHM, "linear"))
         MPIR_Ineighbor_alltoallv_intra_algo_choice = MPIR_INEIGHBOR_ALLTOALLV_INTRA_ALGO_LINEAR;
+    else if (0 == strcmp(MPIR_CVAR_INEIGHBOR_ALLTOALLV_INTRA_ALGORITHM, "gentran_linear"))
+        MPIR_Ineighbor_alltoallv_intra_algo_choice = MPIR_INEIGHBOR_ALLTOALLV_INTRA_ALGO_LINEAR;
```

Comment:
Not "..._ALGO_LINEAR", but "..._ALGO_GENTRAN_LINEAR"

Reply:
Fixed. Thanks for pointing it.